### DISCUSSION TITLE
cue vet on empty file

### DISCUSSION THREAD

**Discussion by hjet** [cue-lang/cue#3362] [Q&A] [answered]
Consider the following example:

`test.cue`
```
#Config: {
  num: int
}
```

`test0.json` (empty file, or file containing whitespace only)
```
     
```

```
cue vet test.cue test0.json -d '#Config'
```
This validates succesfully.

Then
`test1.json`
```
{}
```

```
num: incomplete value int:
    ./test.cue:2:10
```

Should `test0` report a validation error? Is this expected?

Thank you 🙏

**Comment by mvdan** [comment, marked as answer]
The testscript below represents what you describe above:

```
exec cue vet schema.cue data.json -d '#Config'

-- schema.cue --
#Config: {
	num: int
}
-- data.json --
```

Luckily, I also discovered this bug recently while investigating a different issue, and the bug got resolved in master in https://review.gerrithub.io/c/cue-lang/cue/+/1198874 - this fix is included in the latest release candidate. In short, we used to parse `.json` files as if they were `.jsonl`, and an empty `.json` file was then incorrectly decoded as a stream of zero values - which `cue vet` treated as a no-op.

**Reply by hjet** [reply]
Thank you!
